cpp/src/arrow/dataset/test_util.h

```diff
@@ -555,6 +555,21 @@ class FileFormatFixtureMixin : public ::testing::Test {
     ASSERT_OK_AND_ASSIGN(predicate, predicate.Bind(*full_schema));
     ASSERT_FINISHES_OK_AND_EQ(util::nullopt, fragment->CountRows(predicate, options));
   }
+  void TestFragmentEquals() {
+    auto options = std::make_shared<ScanOptions>();
+    auto test_schema = schema({field("f64", float64())});
+    auto reader = this->GetRecordBatchReader(test_schema);
+    auto other_reader = this->GetRecordBatchReader(test_schema);
+    auto source = this->GetFileSource(reader.get());
+    auto other_source = this->GetFileSource(other_reader.get());
+
+    auto fragment = this->MakeFragment(*source);
+    auto other = this->MakeFragment(*other_source);
+
+    EXPECT_EQ(fragment->Equals(*other), false);
+    // TODO: Extend the test cases for Fragment
+    // when ARROW-16855 is merged.
```

Comment:
Is that a TODO for the code here, or for the other PR?

Reply:
Yes. I need to mention this in other PR as well.

Reply:
Sure. I will move it to the other PR